src/internal/connector/onedrive/drive.go

```diff
@@ -343,3 +350,21 @@ func DeleteItem(
 
 	return nil
 }
+
+// driveName returns the drive name for a drive
+// For OneDrive - Graph always sets d.Name to `OneDrive` so
+// instead we parse the webURL and use the last element
+// e.g. if it is "https://<site name>/personal/<user>_onmicrosoft_com/Documents",
+// we return `Documents`
+func driveName(d models.Driveable) string {
+	dt := ptr.Val(d.GetDriveType())
+	switch dt {
+	case oneDriveDriveType:
+		if webURL := ptr.Val(d.GetWebUrl()); webURL != "" {
```

Comment:
nit: I think the standard is to use len checks for empty string?
```suggestion
		if webURL := ptr.Val(d.GetWebUrl()); len(webURL) > 0 {
```